src/core/background/inject.ts

```diff
@@ -0,0 +1,75 @@
+import { getConnectorByUrl } from '@/util/util-connector';
+import browser from 'webextension-polyfill';
+
+/**
+ * Attempts to inject the connector into the page.
+ *
+ * Function does not wait for injection to finish, because it can hang if the tab is asleep.
+ *
+ * @param tab - The tab to inject the connector into
+ * @returns A promise that resolves when the connector is being injected.
+ */
+async function attemptInjectTab(tab: browser.Tabs.Tab) {
+	if (typeof tab.id === 'undefined') {
+		throw new Error('Could not identify tab: ' + tab);
+	}
+
+	let url = tab.url;
+	if (typeof url === 'undefined') {
+		url = await browser.tabs.get(tab.id).then((idTab) => idTab.url);
+	}
+	if (typeof url === 'undefined') {
+		throw new Error('Could not identify URL of tab: ' + tab);
+	}
+
+	return injectConnector(tab.id, url);
+}
+
+/**
+ * Does the actual injection attempt after checking for missing properties.
+ *
+ * @param tabId - The tab to inject the connector into
+ * @param url - The URL of the tab
+ * @returns A promise that resolves when the connector is injected
+ */
+async function injectConnector(tabId: number, url: string) {
+	const connector = await getConnectorByUrl(url);
+
+	if (!connector) {
+		return;
+	}
+
+	/**
+	 * Important note: We do not check if the script already exists here.
+	 * As scripts are always invalidated on reload, and this only runs on install, there is no need.
+	 */
+
+	try {
+		const script = 'content/main.js';
+		browser.scripting.executeScript({
+			target: { tabId },
+			files: [script],
+		});
+	} catch (err) {
+		if (err instanceof Error) {
```

Comment:
Because try catch is not typed (a function could theoretically throw anything and typescript isn't that smart), typescript requires us to validate it.